### PR TITLE
Trainer registration

### PR DESCRIPTION
**Pull request by tnodland**
This PR:
Some stuff is a little messy with the combination of our old schema merging with our new schema, however at the base I think leaving the users controller in for the functionality of registration is a good idea as it until they create an account we don't know what type of user we are dealing with

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class ApplicationController < ActionController::Base
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
```
Comment (VinceCarollo) [inline app/controllers/application_controller.rb]: I dont think we're using this. Can we remove it?

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<%= form_for :user do |f| %>
```
Comment (VinceCarollo) [inline app/views/users/new.html.erb]: Shouldn't we be calling this Client not User?

Comment (kylecornelissen) [reply app/views/users/new.html.erb]: User is so trainers and clients can register using the same form and then it breaks off and creates either a trainer or client.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (kylecornelissen) [reply app/controllers/application_controller.rb]: We will remove it.

**Comment by kylecornelissen** [comment]
Went through the conflicts with Trevor and Vince to fix the broken tests and remove redundant and obsolete files. Everything checks out now so I will merge.

**Pull request merged**